lisa/tools/sysbench.py

```diff
@@ -0,0 +1,563 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+import re
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, cast
+
+from lisa import notifier
+from lisa.executable import Tool
+from lisa.messages import TestStatus, send_sub_test_result_message
+from lisa.operating_system import Posix
+from lisa.util import LisaException
+from lisa.util.process import ExecutableResult
+
+
+@dataclass
+class SysbenchTestResult:
+    exit_code: Optional[int] = 0
+    result: Any = None
+
+
+class Sysbench(Tool):
+    @property
+    def command(self) -> str:
+        return "sysbench"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("sysbench")
+
+        return self._check_exists()
+
+    def __run_sysbench_test(
+        self,
+        test_type: str,
+        threads: int,
+        events: int,
+        verbosity: int,
+        time_limit: int,
+        percentile: int,
+        debug: str,
+        histogram: str,
+        validation: str,
+        extra_args: str = "",
+    ) -> ExecutableResult:
+        args = (
+            f"{test_type} run --threads={threads} --events={events}"
+            f" --validate={validation} --percentile={percentile}"
+            f" --verbosity={verbosity} --histogram={histogram}"
+            f" --time={time_limit} --debug={debug} {extra_args}"
+        )
+
+        result: ExecutableResult = self.run(args)
+        return result
+
+    def run_cpu_perf(
+        self,
+        test_result: Any,
+        subtest_name: str = "",
+        threads: int = 1,
+        events: int = 0,
+        cpu_max_prime: int = 10000,
+        verbosity: int = 5,
+        time_limit: int = 10,
+        percentile: int = 95,
+        debug: str = "on",
+        histogram: str = "on",
+        validation: str = "on",
+    ) -> SysbenchTestResult:
+        extra_args: str = f" --cpu-max-prime={cpu_max_prime}"
+        result: Dict[Any, Any] = {}
+        message: str = ""
+        status = TestStatus.QUEUED
+
+        if not subtest_name:
+            subtest_name = "default_sysbench_cpu_perf_test"
```

Comment:
1. What exact sysbench test executed? Is there a test name from sysbench? If sysbench has only one cpu test, "cpu" is enough. "sysbench" is redundant, because it's in parent test case name.
2. Define the default value on method argument, so it doesn't need to set here.

Reply:
Removed subtest_name.

Using create_perf_message as suggested so there wont be subtest now. We will push multiple result for all iterations for given testcase just like Disk/Network perf test we have.